File: src/web/magmac/app/lang/common/Access.ts
export class Access<V,  R,  T> {
	 Access( variant : V,  receiver : R,  property : String,  maybeArguments : Option<List<T>>) : public {this.variant=variant;this.receiver=receiver;this.property=property;this.maybeArguments=maybeArguments;;}
	public type() : V {return this.variant;;}
	public receiver() : R {return this.receiver;;}
	public property() : String {return this.property;;}
}

File: src/java/magmac/app/lang/common/Access.java
package magmac.app.lang.common;

import magmac.api.Option;
import magmac.api.collect.list.List;

public class Access<V, R, T> {
    private final V variant;
    private final R receiver;
    private final String property;
    private final Option<List<T>> maybeArguments;

    public Access(V variant, R receiver, String property, Option<List<T>> maybeArguments) {
        this.variant = variant;
        this.receiver = receiver;
        this.property = property;
        this.maybeArguments = maybeArguments;
    }

    public V type() {
        return this.variant;
    }

    public R receiver() {
        return this.receiver;
    }

    public String property() {
        return this.property;
    }
}
